### PR TITLE
Now exposing a counter of received messages

### PR DESCRIPTION
**Pull request by zubairov**
Could be useful to see which sensors are sending data and which are not (e.g. when outside of range)

### CODE REVIEW

**Comment by kpetremann** [comment]
you can already do this with existing metrics, like this:

`name: sensor_no_fresh_data
expr: changes(zigbee_temperature[4h]) == 0
for: 15m
labels:
  severity: notification-daily
annotations:
  description: Sensor data did not change for 4 hours: {{ $labels.sensor }}
  sumary: No fresh data (> 4h) for {{ $labels.sensor }}`

File: exporter.py [hunk 1]
```diff
@@ -8,14 +8,19 @@
...
 LOG = logging.getLogger("mqtt-exporter")
 PREFIX = os.environ.get("PROMETHEUS_PREFIX", "mqtt_")
 
 # global variable
 prom_metrics = {}  # pylint: disable=C0103
+topic_label = os.environ.get("TOPIC_LABEL", "topic")
+prom_msg_counter = Counter(
+    f"{PREFIX}message_counter", "Counter of received messages", [topic_label]
+)
+LOG.info("creating counter for message count")
```
Comment (kpetremann) [inline exporter.py:23]: useless: no need for creating a log at the global level

File: exporter.py [hunk 2]
```diff
@@ -8,14 +8,19 @@
...
-from prometheus_client import Gauge, start_http_server
+from prometheus_client import Gauge, Counter, start_http_server
 
 logging.basicConfig(level=os.environ.get("LOG_LEVEL", "INFO"))
 LOG = logging.getLogger("mqtt-exporter")
 PREFIX = os.environ.get("PROMETHEUS_PREFIX", "mqtt_")
 
 # global variable
 prom_metrics = {}  # pylint: disable=C0103
+topic_label = os.environ.get("TOPIC_LABEL", "topic")
```
Comment (kpetremann) [inline exporter.py:19]: when set at global level, a constant should be in uppercase
and please move this line higher (before "global variable" as this is not a variable.

I am surprised pylint did not catch it.

File: exporter.py [hunk 3]
```diff
@@ -52,11 +56,13 @@ def expose_metrics(client, userdata, msg):  # pylint: disable=W0613
             prom_metrics[prom_metric_name] = Gauge(
                 prom_metric_name, "metric generated from MQTT message.", [topic_label]
             )
-            LOG.info("creating prometheus metric: %s", prom_metric_name)
+            LOG.info("creating prometheus metric: %s ", prom_metric_name)
```
Comment (kpetremann) [inline exporter.py:59]: why adding a space? typo?

File: exporter.py [hunk 4]
```diff
@@ -8,14 +8,19 @@
...
 logging.basicConfig(level=os.environ.get("LOG_LEVEL", "INFO"))
 LOG = logging.getLogger("mqtt-exporter")
 PREFIX = os.environ.get("PROMETHEUS_PREFIX", "mqtt_")
 
 # global variable
 prom_metrics = {}  # pylint: disable=C0103
+topic_label = os.environ.get("TOPIC_LABEL", "topic")
+prom_msg_counter = Counter(
+    f"{PREFIX}message_counter", "Counter of received messages", [topic_label]
+)
```
Comment (kpetremann) [inline exporter.py:22]: you may need to add `# pylint: disable=C0103` at the end of the line

**Review by kpetremann — CHANGES_REQUESTED** [review]
Thanks for the PR.

Please update the README to document the new Counter

For next PR, please rebase your branch to upstream before doing the PR to avoid irrelevant commit appearing in the PR.

Also, I'll have a look why the CI did not catch the issues.

File: exporter.py [hunk 5]
```diff
@@ -8,14 +8,19 @@
...
 
 logging.basicConfig(level=os.environ.get("LOG_LEVEL", "INFO"))
 LOG = logging.getLogger("mqtt-exporter")
 PREFIX = os.environ.get("PROMETHEUS_PREFIX", "mqtt_")
 
 # global variable
 prom_metrics = {}  # pylint: disable=C0103
+topic_label = os.environ.get("TOPIC_LABEL", "topic")
+prom_msg_counter = Counter(
+    f"{PREFIX}message_counter", "Counter of received messages", [topic_label]
```
Comment (kpetremann) [inline exporter.py:21]: according Prometheus best practices, label should end with _total for counters:
https://prometheus.io/docs/practices/naming/

**Comment by kpetremann** [comment]
A better way to check if a sensor is not sending data anymore would be to simply add a timestamp in the metrics.

I'll do that after this PR will be merged. A counter is still useful for other things like creating statistics.

File: exporter.py (hunk 5 above)
Comment (zubairov) [reply exporter.py:21]: Yes makes sense to me however the library seems to some naming magic already:

image

Comment (kpetremann) [reply exporter.py]: sure, but better explicit than implicit

Comment (kpetremann) [reply exporter.py]: and message_counter_total is kind of redudant

**Pull request merged**